docs/wiki/ModulesDesign.md

```diff
@@ -474,34 +443,68 @@ Within a bicep file, follow the following conventions:
         properties: {...}
       }
       ```
-- Bicep `modules`:
-  - camel_Snake_Case, i.e `resourceGroup_rbac` ?
-  - File name for nested module is structured as follows: `nested_<resourceName>.bicep` i.e:
-    - `nested_rbac.bicep`
+## Modules
+
+  - Module symbolic names are in camel_Snake_Case, following the schema `<mainResourceType>_<referencedResourceType>` e.g. `storageAccount_fileServices`, `virtualMachine_nic`, `resourceGroup_rbac`.
+  - Modules enable you to reuse code from a Bicep file in other Bicep files. As such they're normally leveraged for deploying child resources (e.g. file services in a storage account), cross referenced resources (e.g. network interface in a virtual machine) or providers (e.g. role assignment in a resource group).
```

Comment:
```suggestion
  - Modules enable you to reuse code from one Bicep file in another Bicep file. As such they're usually leveraged to deploy child resources (e.g. file services in a storage account), cross referenced resources (e.g. network interface in a virtual machine) or providers (e.g. role assignment in a resource group).
```

Reply:
Again there is nothing called providers. Call this extension resources

Reply:
Done.